ocs_ci/ocs/cluster.py

```diff
@@ -924,3 +924,135 @@ def validate_pg_balancer():
+    deployment and cluster expansion tests.
+    This function is specifically for ocs cluster created on 1 AZ VMWare setup
+
+    Args:
+        osd_tree(dict): Dictionary of the values which represent 'osd tree'.
+        number_of_osds(int): total number of osds in the cluster
+
+    Returns:
+        Boolean: True, if the ceph osd tree is formed correctly. Else False
+    """
+    # in case of vmware, there will be only one zone as of now. The OSDs are arranged as follows:
+    # ID  CLASS WEIGHT  TYPE NAME                            STATUS REWEIGHT PRI-AFF
+    # -1       0.99326 root default
+    # -8       0.33109     rack rack0
+    # -7       0.33109         host ocs-deviceset-0-0-dktqc
+    #  1   hdd 0.33109             osd.1                        up  1.00000 1.00000
+    # There will be 3 racks - rack0, rack1, rack2.
+    # When cluster expansion is successfully done, a host and an osd are added in each rack.
+    # The number of hosts will be equal to the number osds the cluster has. Each rack can
+    # have multiple hosts but each host will have only one osd under it.
+    number_of_hosts_expected = number_of_osds/3
+    all_hosts = []
+    racks = osd_tree['nodes'][0]['children']
+
+    for rack in racks:
+        hosts = get_child_nodes_osd_tree(rack, osd_tree)
+        if len(hosts) != number_of_hosts_expected:
+            logging.error(f"Number of hosts under rack {rack} "
+                          f"is not matching the expected ={number_of_hosts_expected} ")
+            return False
+        else:
+            all_hosts.append(hosts)
+
+    all_hosts_flatten = [item for sublist in all_hosts for item in sublist]
+    return check_osds_in_hosts_osd_tree(all_hosts_flatten, osd_tree)
+
+
+def check_osd_tree_3az_aws(osd_tree, number_of_osds):
+    """
+    Checks whether an OSD tree is created/modified correctly. This can be used as a verification step for
+    deployment and cluster expansion tests.
+    This function is specifically for ocs cluster created on 3 AZ AWS config
+
+    Args:
+        osd_tree(dict): Dictionary of the values which represent 'osd tree'.
+        number_of_osds(int): total number of osds in the cluster
+
+    Returns:
+        Boolean: True, if the ceph osd tree is formed correctly. Else False
+    """
+    all_hosts = []
+    region = osd_tree['nodes'][0]['children']
+
+    zones = get_child_nodes_osd_tree(region[0], osd_tree)
+    for each_zone in zones:
+        hosts_in_each_zone = get_child_nodes_osd_tree(each_zone, osd_tree)
+        if len(hosts_in_each_zone) != number_of_osds / 3:  # 3 is replica_factor
+            logging.error(f"number of hosts in zone is incorrect")
+            return False
+        else:
+            all_hosts.append(hosts_in_each_zone)
+
+    all_hosts_flatten = [item for sublist in all_hosts for item in sublist]
+
+    return check_osds_in_hosts_osd_tree(all_hosts_flatten, osd_tree)
+
+
+def check_osd_tree_1az_aws(osd_tree, number_of_osds):
```

Comment:
I think I commented about it in the previous PR, but I think you should consolidate this function and the one above

Reply:
output of 'ceph osd tree' is completely different for 1 az  and 3az. So @hnallurv and I agreed for different functions